### DISCUSSION TITLE
How to use getFieldValue inside a Form.List?

### DISCUSSION THREAD

**Discussion by Thiht** [ant-design/ant-design#32294] [Q&A] [answered]
Hi,
I've created a Form.List like this: (it's basically a copy paste of https://ant.design/components/form/#components-form-demo-dynamic-form-items)

```js
const KeyValueEditor = ({ name }): JSX.Element => (
  <Form.List name={name}>
    {(fields, { add, remove }) => (
      <>
        {fields.map(({ key, name, fieldKey, ...restField }) => (
          <Space key={key}>
            <Form.Item
              {...restField}
              name={[name, "key"]}
              fieldKey={[fieldKey, "key"]}
            >
              <Input placeholder="Key" />
            </Form.Item>

            <Form.Item
              {...restField}
              name={[name, "matcher"]}
              fieldKey={[fieldKey, "matcher"]}
            >
              <Select>
                <Select.Option value="ShouldEqual">ShouldEqual</Select.Option>
                <Select.Option value="ShouldBeEmpty">
                  ShouldBeEmpty
                </Select.Option>
              </Select>
            </Form.Item>

            <Form.Item
              {...restField}
              name={[name, "value"]}
              fieldKey={[fieldKey, "value"]}
            >
              <Input placeholder="Value" />
            </Form.Item>

            <MinusCircleOutlined onClick={() => remove(name)} />
          </Space>
        ))}

        <Form.Item>
          <Button onClick={() => add({})} icon={<PlusOutlined />}>
            Add field
          </Button>
        </Form.Item>
      </>
    )}
  </Form.List>
);
```

Now I want the third field to disappear if the second field of the same row contains "ShouldBeEmpty", so I tried to replace the third field with something like this:

```js
<Form.Item noStyle>
  {({ getFieldValue,  }) => (
    getFieldValue([name, fieldKey, "matcher"]) !== "ShouldBeEmpty && (
      <Form.Item
        {...restField}
         name={[name, "value"]}
         fieldKey={[fieldKey, "value"]}
      >
         <Input placeholder="Value" />
      </Form.Item>
    )
  )}
</Form.Item>
```

but I can't figure out what NamePath to use in my getFieldValue. Do you have an idea?

**Comment by Thiht** [comment, marked as answer]
I struggled with this for hours, and 10 minutes after posting this question I found my answer... In my code I shadow the `name` variable at line 4 so i don't reference the correct path in my `getFieldValue`. With the correct variable it works fine.

**Reply by JordanLavell** [reply]
What is the correct variable?

**Reply by usersina** [reply]
```javascript 
form.getFieldValue(['form-list-name', field.name, 'input-name'])
```

**Reply by alizain143** [reply]
how can i use form.getFieldValue if a formlist is nested inside a formlist and i want to get a value from the nested form list ?

**Reply by nouman2075** [reply]
> how can i use form.getFieldValue if a formlist is nested inside a formlist and i want to get a value from the nested form list ?

```
form.getFieldValue(['first-list-name', field.name, 'second-list-name', 'firstName' ])
```

**Comment by zxlvera** [comment]
Sorry to hijack an old thread, but i couldn't figure out how to read the value of another field using `getFieldValue` inside `Form.List`. In the screenshot, I referenced the first field "name" but `<p>hello</p>` does not render.

![image](https://github.com/ant-design/ant-design/assets/17290983/c37578d5-52a8-44c5-b7c1-103c538748d0)

**Reply by nouman2075** [reply]
Pass `shouldUpdate` to `<Form.Item noStyle ...>`.

**Reply by zxlvera** [reply]
Thank you! I passed `shouldUpdate` without declaring a boolean and it wasn't working:
```
<Form.Item noStyle shouldUpdate>{...}</Form.Item>
```
but this works
```
<Form.Item noStyle shouldUpdate={true}>{...}</Form.Item>
```

**Comment by Shruti271099** [comment]
{form.getFieldValue(["inputFields", name, "type"]) ===
                            "Title Company" && (
                            <Form.Item
                              {...restField}
                              name={[name, "title Company"]}
                              rules={[
                                {
                                  required: true,
                                  message: "Missing titleCompany",
                                },
                              ]}>
                              {console.log("in")}
                              <Input placeholder="titleCompany" />
                            </Form.Item>
                          )}
                          
                          use form.getFieldValue instead of {({getFieldValue}) => getFieldValue(...))}
